components/backends/trtllm/src/dynamo/verifier/main.py

```diff
@@ -0,0 +1,207 @@
+
+    config = cmd_line_args()
+    await init(runtime, config)
+
+
+async def init(runtime: DistributedRuntime, config: Config):
+    """
+    Instantiate and serve
+    """
+    logging.info(f"Initializing the worker with config: {config}")
+
+    next_client = None
+    if config.next_endpoint:
+        logging.info(
+            f"Initializing next worker client for endpoint: {config.next_endpoint}"
+        )
+        parsed_namespace, parsed_component_name, parsed_endpoint_name = parse_endpoint(
+            config.next_endpoint
+        )
+        next_client = (
+            await runtime.namespace(parsed_namespace)
+            .component(parsed_component_name)
+            .endpoint(parsed_endpoint_name)
+            .client()
+        )
+
+    component = runtime.namespace(config.namespace).component(config.component)
+    await component.create_service()
+
+    # Convert model path to Path object if it's a local path, otherwise keep as string
+    model_path = str(config.model_path)
+
+    arg_map = {
+        "model": model_path,
+        "tensor_parallel_size": config.tensor_parallel_size,
+        "backend": "pytorch",
+        "skip_tokenizer_init": True,
+    }
+    if config.extra_engine_args != "":
+        # TODO: Support extra engine args from json file as well.
+        arg_map = update_llm_args_with_extra_options(arg_map, config.extra_engine_args)
+    if config.publish_events_and_metrics:
+        # 'event_buffer_max_size' is required to enable TRTLLM to publish kv cache events.
+        kv_cache_config = None
+        if "kv_cache_config" not in arg_map:
+            kv_cache_config = {}
+            kv_cache_config["event_buffer_max_size"] = DEFAULT_KV_EVENT_BUFFER_MAX_SIZE
+        else:
+            kv_cache_config = arg_map["kv_cache_config"]
+            if not kv_cache_config.event_buffer_max_size:
+                kv_cache_config.event_buffer_max_size = DEFAULT_KV_EVENT_BUFFER_MAX_SIZE
+        arg_map["kv_cache_config"] = kv_cache_config
+
+        # Only pytorch backend is supported for now to publish events and metrics.
+        if "backend" not in arg_map:
+            arg_map["backend"] = "pytorch"
+        elif arg_map["backend"] != "pytorch":
+            logging.error(
+                "Only pytorch backend is supported for now to publish events and metrics."
+            )
+            sys.exit(1)
+
+    # Set speculative_config to use DynamoAPIDrafter in the verifier worker
+    try:
+        drafter_config = arg_map["drafter_config"]
+        drafter_endpoint = drafter_config["drafter_endpoint"]
+        if "max_draft_len" in drafter_config:
+            max_draft_len = drafter_config["max_draft_len"]
+        else:
+            max_draft_len = 10
+    except KeyError:
+        raise ValueError(
+            "Verifier worker requires drafter_config to be specified in the engine config with drafter_endpoint."
+        )
+    del arg_map["drafter_config"]
+
+    drafter_config = ExternalAPIConfig(
```

Comment:
We should not create new files just to support extra configs. Let's adapt the dynamo.trtllm itself.